modules/apps/oauth2-provider/oauth2-provider-scope-impl/src/main/java/com/liferay/oauth2/provider/scope/internal/ThreadLocalScopeContextScopeChecker.java

```diff
@@ -107,10 +98,7 @@ public boolean checkScope(String scope) {
 		}
 
 		Collection<OAuth2ScopeGrant> oAuth2ScopeGrants =
-			_oAuth2ScopeGrantLocalService.getOAuth2ScopeGrants(
-				_companyIdThreadLocal.get(), _applicationNameThreadLocal.get(),
-				_bundleSymbolicNameThreadLocal.get(),
-				_accessTokenThreadLocal.get());
+			_getOAuth2ScopeGrants();
```

Comment:
@LisztLiszt After extracting the methods, please also check if you can inline the variable.